contracts/interchain-token-service/tests/deploy_interchain_token.rs

```diff
@@ -0,0 +1,222 @@
+use axelar_gas_service::{AxelarGasService, AxelarGasServiceClient};
+use axelar_gateway::testutils;
+use axelar_gateway::AxelarGatewayClient;
+use axelar_soroban_std::assert_contract_err;
+use axelar_soroban_std::assert_invoke_auth_err;
+use interchain_token::InterchainTokenClient;
+use interchain_token_service::error::ContractError;
+use interchain_token_service::{InterchainTokenService, InterchainTokenServiceClient};
+
+use soroban_sdk::BytesN;
+use soroban_sdk::IntoVal;
+use soroban_sdk::{testutils::Address as _, Address, Env, String};
+use soroban_token_sdk::metadata::TokenMetadata;
+
+const INTERCHAIN_TOKEN_WASM_HASH: &[u8] = include_bytes!("testdata/interchain_token.wasm");
+
+fn setup_token_metadata(env: &Env, name: &str, symbol: &str, decimal: u32) -> TokenMetadata {
+    TokenMetadata {
+        decimal,
+        name: name.into_val(env),
+        symbol: symbol.into_val(env),
+    }
+}
+
+fn setup_gateway<'a>(env: &Env) -> AxelarGatewayClient<'a> {
+    let (_, client) = testutils::setup_gateway(env, 0, 5);
+    client
+}
+
+fn setup_gas_service<'a>(env: &Env) -> AxelarGasServiceClient<'a> {
+    let owner: Address = Address::generate(env);
+    let gas_collector: Address = Address::generate(env);
+    let gas_service_id = env.register(AxelarGasService, (&owner, &gas_collector));
+    let gas_service_client = AxelarGasServiceClient::new(env, &gas_service_id);
+
+    gas_service_client
+}
+
+fn setup_its(env: &Env) -> InterchainTokenServiceClient {
```

Comment:
why is this being duplicated? Use `utils/mod.rs`

Reply:
updated